### PR TITLE
octavePackages.symbolic: unstable-2021-10-16 -> 3.0.1

### PR DESCRIPTION
**Pull request by Tungsten842**
###### Description of changes
https://github.com/cbm755/octsympy/releases/tag/v3.0.1
Symbolic is currently broken, updating it makes it work.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ravenjoad — APPROVED** [review]
Looks good to me! I'm glad to see that symbolic finally got an update.

File: pkgs/development/octave-modules/symbolic/default.nix [hunk 1]
```diff
@@ -1,27 +1,26 @@
...
+    ps.sympy
+    ps.mpmath
+  ]);
 
-in buildOctavePackage rec {
+in
+buildOctavePackage rec {
   pname = "symbolic";
-  version = "unstable-2021-10-16";
+  version = "v3.0.1";
```
Comment (SuperSandro2000) [inline pkgs/development/octave-modules/symbolic/default.nix:17]:
```suggestion
  version = "3.0.1";
```

File: pkgs/development/octave-modules/symbolic/default.nix [hunk 2]
```diff
@@ -1,27 +1,26 @@
...
 
-  # https://github.com/cbm755/octsympy/issues/1023 has been resolved, however
-  # a new release has not been made
   src = fetchFromGitHub {
     owner = "cbm755";
     repo = "octsympy";
-    rev = "5b58530f4ada78c759829ae703a0e5d9832c32d4";
-    sha256 = "sha256-n6P1Swjl4RfgxfLY0ZuN3pcL8PcoknA6yxbnw96OZ2k=";
+    rev = version;
```
Comment (SuperSandro2000) [inline pkgs/development/octave-modules/symbolic/default.nix:22]:
```suggestion
    rev = "${version}";
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/development/octave-modules/symbolic/default.nix (hunk 2 above)
Comment (KarlJoad) [reply pkgs/development/octave-modules/symbolic/default.nix:22]: The `rev` field does need the `v` at the front of the version. The release tag is `v3.0.1` upstream.

**Review by ravenjoad — APPROVED** [review]

**Review by ravenjoad — APPROVED** [review]

**Comment by QuantMint** [comment]
@SuperSandro2000

File: pkgs/development/octave-modules/symbolic/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/development/octave-modules/symbolic/default.nix]: We strip that from version. See the contributing guide.

**Pull request merged**